### PR TITLE
update compress support compressindex

### PR DESCRIPTION
**Pull request by owen200008**
update compress support compressindex

### CODE REVIEW

File: extensions/assets-manager/AssetsManagerEx.h [hunk 1]
```diff
@@ -43,6 +43,17 @@
 
 NS_CC_EXT_BEGIN
 
+//! All files to be decompressed
+struct CompressedFilesInfoAsserts
+{
+    std::string     m_strFileName;
```
Comment (minggo) [inline extensions/assets-manager/AssetsManagerEx.h]: Please do the change `m_strFileName -> _fileName` to obey the coding style.

File: extensions/assets-manager/AssetsManagerEx.h [hunk 2]
```diff
@@ -43,6 +43,17 @@
...
+    int             m_nCompressIndex;
+    CompressedFilesInfoAsserts()
```
Comment (minggo) [inline extensions/assets-manager/AssetsManagerEx.h]: CompressedFilesInfoAsserts() : xxx() {};

please initialize like above.

**Comment by minggo** [comment]
this PR looks good to me, @pandamicro could you please take a look too.

File: extensions/assets-manager/Manifest.cpp [hunk 3]
```diff
@@ -392,6 +398,7 @@ Manifest::Asset Manifest::parseAsset(const std::string &path, const rapidjson::V
     }
     else asset.downloadState = DownloadState::UNSTARTED;
     
+
```
Comment (minggo) [inline extensions/assets-manager/Manifest.cpp]: please remove unneeded space

File: extensions/assets-manager/Manifest.h [hunk 4]
```diff
@@ -72,6 +72,12 @@ class CC_EX_DLL Manifest : public Ref
         std::string path;
         bool compressed;
         DownloadState downloadState;
+        int m_nCompressIndex;
+        Asset()
+        {
+            compressed = false;
+            m_nCompressIndex = 0;
+        }
```
Comment (minggo) [inline extensions/assets-manager/Manifest.h]: m_nCompressIndex -> compressIndex
and please use parameter initialization list

Comment (minggo) [reply extensions/assets-manager/Manifest.h]: And should initialize `downloadState ` too?

File: extensions/assets-manager/AssetsManagerEx.h [hunk 5]
```diff
@@ -43,6 +43,15 @@
 
 NS_CC_EXT_BEGIN
 
+//! All files to be decompressed
+struct CompressedFilesInfoAsserts
+{
+    std::string     _fileName;
+    int             _nCompressIndex;
```
Comment (minggo) [inline extensions/assets-manager/AssetsManagerEx.h:50]: i think it is better to rename to _compressIndex, `n` is used together with `m_`.

**Comment by pandamicro** [comment]
@owen200008 Please take a look at this PR

https://github.com/cocos2d/cocos2d-x/pull/17115

I have fundamentally refactored AssetsManagerEx, including the decompress process, it's now much more secured, I think your PR won't be needed any more, can you confirm that ?

Thank you anyway for your effort

**Pull request closed without merging**